### PR TITLE
fix the security issue and the duplicate users

### PR DESCRIPTION
**Pull request by Abdulhafeez012**
- fix issue #33 
- fix the duplicate user

### CODE REVIEW

**Review by zAbuQasem — CHANGES_REQUESTED** [review]
```py
file = os.path.splitext(os.path.basename(uploaded_file.name))[0]
file_path = os.path.join(settings.MEDIA_ROOT, upload_directory, file)
```

**Comment by zAbuQasem** [comment]
@Abdulhafeez012 Please checkout my change request

**Comment by zAbuQasem** [comment]
Please make sure to whitelist extensions before saving the file, it doesn't introduce an issue in our case here but worth doing

**Comment by Abdulhafeez012** [comment]
> ```python
> file = os.path.splitext(os.path.basename(uploaded_file.name))[0]
> file_path = os.path.join(settings.MEDIA_ROOT, upload_directory, file)
> ```

Done

**Comment by zAbuQasem** [comment]
@Abdulhafeez012 Are you going to work on extension whitelist for the next release?

File: apps/post/utils.py [hunk 1]
```diff
@@ -7,8 +7,10 @@ def handle_uploaded_file(uploaded_file):
     # Define the directory where you want to save the file, relative to your media root.
     upload_directory = 'uploads/'  # This is a subdirectory within the media root.
 
+    file = os.path.splitext(os.path.basename(uploaded_file))[0]
```
Comment (zAbuQasem) [inline apps/post/utils.py:10]: Shouldn't be this `uploaded_file.name` ?

Comment (Abdulhafeez012) [reply apps/post/utils.py:10]: nope , casue the uploaded_file is come from request.FILE.get('media_file') 
uploaded_file e.g image.png

**Comment by Abdulhafeez012** [comment]
> @Abdulhafeez012 Are you going to work on an extension whitelist for the next release?

next release, to make a list of extensions, after we know the specific extensions we use

**Pull request merged**